cuda/test/matrix/dense_kernels.cpp

```diff
@@ -609,4 +609,37 @@ TEST_F(Dense, OutplaceAbsoluteMatrixIsEquivalentToRef)
 }
 
 
+TEST_F(Dense, MakeComplexIsEquivalentToRef)
+{
+    set_up_apply_data();
+
+    auto complex_x = x->make_complex();
+    auto dcomplex_x = dx->make_complex();
+
+    GKO_ASSERT_MTX_NEAR(complex_x, dcomplex_x, 1e-14);
```

Comment:
Could you try testing for exact equality? We shouldn't be doing any conversions here.
```suggestion
    GKO_ASSERT_MTX_NEAR(complex_x, dcomplex_x, 0.0);
```